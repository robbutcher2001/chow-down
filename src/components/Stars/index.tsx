import React, { ReactNode } from 'react';

import styled from 'styled-components';

interface StarsProps {
  rating: number
};

const Stars = styled.section`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  font-size: 1.2rem;
  padding: 0.5rem;

  > mark {
    color: #4acaa8;
    background-color: rgba(0, 0, 0, 0);
  }
`

// TODO: convert to Unicode stars and use CSS custom var (--rating) to colour: https://css-tricks.com/five-methods-for-five-star-ratings/
export default (props: StarsProps) => (
  <Stars>
    {function () {
      const rating: ReactNode[] = [];
      for (let i = 0; i < props.rating; i++) {
        rating.push(<mark key={i} >&#9733;</mark>);
      }
      return rating;
    }()}
  </Stars>
);