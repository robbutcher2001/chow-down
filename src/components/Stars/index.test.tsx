import React from 'react';
import renderer from 'react-test-renderer';

import Stars from '.';

const mockRating = {
  rating: 4
};

test('Stars basic snapshot render', () => {
  const stars = renderer.create(
    <Stars {...mockRating} />
  );

  expect(stars.toJSON()).toMatchSnapshot();
});