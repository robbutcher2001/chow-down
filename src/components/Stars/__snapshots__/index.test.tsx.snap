// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars basic snapshot render 1`] = `
<section
  className="sc-AxjAm jseqAN"
>
  <mark>
    ★
  </mark>
  <mark>
    ★
  </mark>
  <mark>
    ★
  </mark>
  <mark>
    ★
  </mark>
</section>
`;
